lib/internal/mime.js

```diff
@@ -320,10 +442,11 @@ class MIMEType {
   set type(v) {
     v = `${v}`;
     const invalidTypeIndex = SafeStringPrototypeSearch(v, NOT_HTTP_TOKEN_CODE_POINT);
-    if (v.length === 0 || invalidTypeIndex !== -1) {
+    if (invalidTypeIndex !== -1 || v.length === 0) {
```

Comment:
is this maybe better?

```js
    if (v.length === 0) {
      throw new ERR_INVALID_MIME_SYNTAX('type', v, -1);
    }

    const invalidTypeIndex = SafeStringPrototypeSearch(v, NOT_HTTP_TOKEN_CODE_POINT);
    if (invalidTypeIndex !== -1) {
      throw new ERR_INVALID_MIME_SYNTAX('type', v, invalidTypeIndex);
    }
```

Reply:
We can even merge the `SafeStringPrototypeSearch` within the `if` statement

Reply:
If you want to pass the invalidTypeIndex to the error, then you have to do it outside the if